From a66adf2618d8d92b80c79537c7bcaaedea2bd9a4 Mon Sep 17 00:00:00 2001
From: Alexander Bokovoy <abokovoy@redhat.com>
Date: Tue, 29 Apr 2025 09:37:44 +0300
Subject: [PATCH] freeipa.spec.in: update BIND-related dependencies

BIND in Fedora 42+ includes a custom backport for DNSSEC support when
using OpenSSL provider API. Make sure we have that support included.

For RHEL 10 we should be using a similar build but it is not yet
available, so make sure we include the version that is up to date prior
to enabling DNSSEC with OpenSSL provider API. Once new BIND build is
available, we can enable OpenSSL provider API usage in ipaplatform.rhel.

Fixes: https://pagure.io/freeipa/issue/9696

Signed-off-by: Alexander Bokovoy <abokovoy@redhat.com>
Reviewed-By: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: Alexander Bokovoy <abokovoy@redhat.com>
Reviewed-By: Rob Crittenden <rcritten@redhat.com>
---
 freeipa.spec.in | 21 ++++++++++++++++-----
 1 file changed, 16 insertions(+), 5 deletions(-)

diff --git a/freeipa.spec.in b/freeipa.spec.in
index 78004dc4fcec87079efcd235dcbf61ae2c20c669..78b044b026de6181264a3572779596325af89158 100755
--- a/freeipa.spec.in
+++ b/freeipa.spec.in
@@ -90,7 +90,13 @@
 
 # Fix for TLS 1.3 PHA, RHBZ#1775158
 %global httpd_version 2.4.37-21
+
+# DNSSEC support with OpenSSL provider API in RHEL 10
+%if 0%{?rhel} < 10
 %global bind_version 9.11.20-6
+%else
+%global bind_version 9.18.33-3
+%endif
 
 # support for passkey
 %global sssd_version 2.9.0
@@ -139,11 +145,11 @@
 # Fix for TLS 1.3 PHA, RHBZ#1775146
 %global httpd_version 2.4.41-9
 
-# Fix for RHBZ#2117342
-%if 0%{?fedora} < 37
-%global bind_version 9.11.24-1
+%if 0%{?fedora} < 42
+%global bind_version 32:9.18.33-1
 %else
-%global bind_version 32:9.18.7-1
+# BIND version with backport of DNSSEC support over OpenSSL provider API
+%global bind_version 32:9.18.35-2
 %endif
 # Don't use Fedora's Python dependency generator on Fedora 30/rawhide yet.
 # Some packages don't provide new dist aliases.
@@ -626,7 +632,12 @@ If you are installing an IPA server, you need to install this package.
 Summary: IPA integrated DNS server with support for automatic DNSSEC signing
 BuildArch: noarch
 Requires: %{name}-server = %{version}-%{release}
-Requires: bind-dyndb-ldap >= 11.2-2
+# Both Fedora 42+ and RHEL support newer bind-dyndb-ldap 11.11
+%if 0%{?fedora} < 42
+Requires: bind-dyndb-ldap >= 11.10-33
+%else
+Requires: bind-dyndb-ldap >= 11.11
+%endif
 Requires: bind >= %{bind_version}
 Requires: bind-utils >= %{bind_version}
 # bind-dnssec-utils is required by the OpenDNSSec integration
-- 
2.49.0

